core/src/main/scala/org/ensime/indexer/lucene/SimpleLucene.scala

```diff
@@ -108,24 +119,42 @@ class SimpleLucene(path: Path, analyzers: Map[String, Analyzer]) extends SLF4JLo
    * 10,000 deletes to take about 1 second and inserts about 100ms.
    * Each call to this method constitutes a batch UPDATE operation.
    */
-  def update(delete: Seq[Query], create: Seq[Document], commit: Boolean = true): Unit = {
-    if (delete.nonEmpty) {
-      writer.deleteDocuments(delete.toArray: _*)
-      if (commit) writer.commit()
-    }
-
-    if (create.nonEmpty) {
-      create foreach { doc =>
-        writer addDocument doc
+  def update(delete: Seq[Query], create: Seq[Document], commit: Boolean = true): Future[Unit] =
```

Comment:
if you add the `Future` onto the method declaration line, you can dramatically reduce the diff size (and then avoid getting the blame for writing the code in here!)